frontend/common/stores/project-store.js

```diff
@@ -84,15 +84,22 @@ const controller = {
 
   editEnv: (env) => {
     API.trackEvent(Constants.events.EDIT_ENVIRONMENT)
-    data.put(`${Project.api}environments/${env.api_key}/`, env).then((res) => {
-      const index = _.findIndex(store.model.environments, { id: env.id })
-      store.model.environments[index] = res
-      store.saved()
-      getStore().dispatch(
-        environmentService.util.invalidateTags(['Environment']),
-      )
-      AppActions.refreshOrganisation()
-    })
+    data
+      .put(`${Project.api}environments/${env.api_key}/`, env)
+      .then((res) => {
+        const index = _.findIndex(store.model.environments, { id: env.id })
+        store.model.environments[index] = res
+        store.saved()
+        getStore().dispatch(
+          environmentService.util.invalidateTags(['Environment']),
+        )
+        AppActions.refreshOrganisation()
+      })
+      .catch((e) => {
+        e.json().then((result) => {
+          toast(result.metadata[0], 'danger')
```

Comment:
This assumes result.metadata isn't null.

Reply:
Corrected